backend/src/test/java/ca/mcgill/purposeful/features/ID015_createIdeaStepDefinitions.java

```diff
@@ -0,0 +1,233 @@
+
+  @Autowired IdeaRepository ideaRepository;
+
+  @Autowired URLRepository urlRepository;
+
+  @Autowired private TestRestTemplate client;
+
+  @Autowired private CucumberUtil cucumberUtil;
+
+  private String jwtToken;
+
+  private ResponseEntity<?> response;
+
+  @Given("the database contains the following user account:")
+  public void the_database_contains_the_following_user_account(DataTable dataTable) {
+    // create app user
+    appUserService.registerRegularUser(
+        dataTable.cell(1, 2), dataTable.cell(1, 3), dataTable.cell(1, 0), dataTable.cell(1, 1));
+  }
+
+  @Given("the number of ideas in the database is {int}")
+  public void the_number_of_ideas_in_the_database_is(int int1) {
+    assertEquals(ideaRepository.count(), int1);
+  }
+
+  @Given("the database contains the following domains:")
+  public void the_database_contains_the_following_domains(DataTable dataTable) {
+    for (String domain_name : dataTable.asList()) {
+      Domain domain = new Domain();
+      domain.setName(domain_name);
+      domainRepository.save(domain);
+    }
+  }
+
+  @Given("the database contains the following topics:")
+  public void the_database_contains_the_following_topics(DataTable dataTable) {
+    for (String topic_name : dataTable.asList()) {
+      Topic topic = new Topic();
+      topic.setName(topic_name);
+      topicRepository.save(topic);
+    }
+  }
+
+  @Given("the database contains the following techs:")
+  public void the_database_contains_the_following_techs(DataTable dataTable) {
+    for (String tech_name : dataTable.asList()) {
+      Technology tech = new Technology();
+      tech.setName(tech_name);
+      technologyRepository.save(tech);
+    }
+  }
+
+  @Given("the database contains the following URLs:")
+  public void the_database_contains_the_following_ur_ls(DataTable dataTable) {
+    for (String url_str : dataTable.asList()) {
+      URL url = new URL();
+      url.setURL(url_str);
+      url.setPresetIcon(true);
+      urlRepository.save(url);
+    }
+  }
+
+  @Given("that the user is logged in with the email {string} and the password {string}")
+  public void that_the_user_is_logged_in_with_the_email_and_the_password(
+      String string, String string2) {
+    if (SecurityContextHolder.getContext().getAuthentication() == null) {
+      // Login as the user
+      HttpEntity<String> requestEntity =
+          new HttpEntity<>(cucumberUtil.basicAuthHeader(string, string2));
+      this.response = client.exchange("/login", HttpMethod.POST, requestEntity, String.class);
+
+      // check that the login was successful
+      assertEquals(200, this.response.getStatusCode().value());
+
+      // save the jwt token
+      this.jwtToken = this.response.getBody().toString();
+    }
+  }
+
+  @When(
+      "^the user creates an idea with title (.*), purpose (.*), description (.*), paid status (.*), progress status (.*), private status (.*), domains (.*), topics (.*), techs (.*), icon (.*), supporting images (.*)$")
```

Comment:
Could you be more consistent in the definition of the line from the feature file, right now you are using two different formats

Reply:
I just committed all the changes you suggested, but i didn't understand what you mean by this comment, could you explain ? Thanks

Reply:
I fixed that issue, you don't need to worry about that anymore.

You have not resolved this comment yet: https://github.com/WassimJabz/Purposeful/pull/152#discussion_r1117837034